lib/spack/spack/cmd/testsuite.py

```diff
@@ -0,0 +1,221 @@
+        cdash = '--log-format=cdash-simple'
+    print args.complete
+    enabledTests = []
+    compilers = []
+    packages = []
+    tests = []
+    testPackages = []
+    versions = []
+    packageVersion = []
+    compilerVersion = []
+    path = ""
+    dashboards = []
+    #designed to use a single file and modify the enabled tests, thus requiring a single file modification.
+    for files in args.yamlFile: #read yaml files which contains description of tests
+        with open(files, 'r') as stream:
+            try:
+                yamlFile= yaml.load(stream)
+            except yaml.YAMLError as exc:
+                print(exc)
+
+            enabledTests= yamlFile['enable']
+            packages= yamlFile['packages'] # list of packages to test. Current libelf, libdwarf, bzip2
+            compilers = yamlFile['compilers']
+            
+            try:
+                path=yamlFile['path']
+            except KeyError as e:
+                tty.msg("Path not found, will use " + str(spack.prefix)+"/var/spack/cdash/")
+
+            try:
+                dashboards = yamlFile['dashboard']
+            except KeyError as e:
+                if path is "":
+                    tty.msg("dashboard url was not found. xml files will be stored at " + str(spack.prefix)+"/var/spack/cdash/")
+                else:
+                    tty.msg("dashboard url was not found. xml files will be stored at path provided.")
+    #creating a list of packages 
+    for package in packages:
+        for pkg in package:
+            if pkg in enabledTests:
+                versions = package[pkg][0]['versions']
+                for version in versions:
+                    # producing packages at available versions. Sample file contains only checksum'd
+                    packageVersion.append(str(pkg)+"@"+str(version))
+
+    #creating a list of compilers
+    for compiler in compilers:
+        for comp in compiler:
+            versions = compiler[comp][0]['versions']
+            for version in versions:
+                # producing compilers at available versions. Sample file contains only checksum'd
+                compilerVersion.append(str(comp)+"@"+str(version))
+    #reducing compiler list to whats actually available on the system
+    compilerVersion = reduceCompilerList(compilerVersion)
+    #producing a list of tests with a combination of packages and compilers
+    for pkg in packageVersion:
+        for comp in compilerVersion:
+            tests.append(str(pkg)+"%"+str(comp))
+
+    #loading test excusions
+    exclusions = yamlFile['exclusions']
+    if len(exclusions) != 0:
+        #remove test that match the exclusion
+        for exclusion in exclusions:
+            tests=removeTests(tests, exclusion)
+
+
+    concreteTests = []
+    
+    #setting up tests for contretizing
+    for test in tests:
+        spec = Spec(test)
+        #uninstall all packages before installing. This will reduce the number of skipped package installs.
```

Comment:
We should figure out how to make this nondestructive.   I think the `test-suite` command should probably make its own, empty package database and directory layout, separate from the ones used by Spack.  You can look at the tests, which create a mock database and mock directory layout, to figure out how to do that.  @becker33 and @scheibelp can also help out with this.